src/extensions/arith.rs

```diff
@@ -60,59 +62,66 @@ pub enum ExprInner {
     /* Two children */
     /// Add two Arith expressions.
     /// `[X] [Y] ADD64 <1> EQUALVERIFY`
-    Add(Box<Expr>, Box<Expr>),
+    Add(Box<Expr<T>>, Box<Expr<T>>),
     /// Subtract (X-Y)
     /// `[X] [Y] SUB64 <1> EQUALVERIFY`
-    Sub(Box<Expr>, Box<Expr>),
+    Sub(Box<Expr<T>>, Box<Expr<T>>),
     /// Multiply two Expr expressions. (a*b)
     /// `[X] [Y] MUL64 <1> EQUALVERIFY`
-    Mul(Box<Expr>, Box<Expr>),
+    Mul(Box<Expr<T>>, Box<Expr<T>>),
     /// Divide two Expr expressions. (a//b)
     /// The division operation pushes the quotient(a//b) such that the remainder a%b
     /// (must be non-negative and less than |b|).
     /// `[X] [Y] DIV64 <1> EQUALVERIFY NIP`
-    Div(Box<Expr>, Box<Expr>),
+    Div(Box<Expr<T>>, Box<Expr<T>>),
     /// Modulo operation (a % b)
     /// The division operation the remainder a%b (must be non-negative and less than |b|).
     /// `[X] [Y] DIV64 <1> EQUALVERIFY DROP`
-    Mod(Box<Expr>, Box<Expr>),
+    Mod(Box<Expr<T>>, Box<Expr<T>>),
     /// BitWise And (a & b)
     /// `[X] [Y] AND` (cannot fail)
-    BitAnd(Box<Expr>, Box<Expr>),
+    BitAnd(Box<Expr<T>>, Box<Expr<T>>),
     /// BitWise or (a | b)
     /// `[X] [Y] OR` (cannot fail)
-    BitOr(Box<Expr>, Box<Expr>),
+    BitOr(Box<Expr<T>>, Box<Expr<T>>),
     /// BitWise or (a ^ b)
     /// `[X] [Y] XOR` (cannot fail)
-    Xor(Box<Expr>, Box<Expr>),
+    Xor(Box<Expr<T>>, Box<Expr<T>>),
     /* One child*/
     /// BitWise invert (!a)
     /// `[X] INVERT` (cannot fail)
-    Invert(Box<Expr>),
+    Invert(Box<Expr<T>>),
     /// Negate -a
     /// `[X] NEG64 <1> EQUALVERIFY`
-    Negate(Box<Expr>),
+    Negate(Box<Expr<T>>),
+
+    /// Push the price as LE64 signed from oracle.
+    /// `2DUP TOALTSTACK <T> OP_GREATERTHANEQ VERIFY CAT SHA256 <K> CHECKSIGFROMSTACKVERIFY OP_FROMATLSTACK`
+    /// The fragment checks that the input timestamp is less than time at which the price was signed with
+    /// the given oracle key. The asset of which price is being checked is implicitly decided by the
+    /// public key
+    PriceOracle1(T, u64),
```

Comment:
nit: in b720130861a0505d2252623383b64f384acd5277 `Pk` would be a much clearer name for this generic (but I won't make you go change it :P)

Also I'd kinda prefer if `PriceOracle1`'s fields have names, but again, no big deal.

Reply:
Or maybe it should be called `Ext`

Reply:
Stuck to T because other fragments in `AssetExpr`, `SpkExpr`, `ValueExpr` also use T